core/src/main/java/io/grpc/ServerServiceDefinition.java

```diff
@@ -126,6 +126,21 @@ private Builder(ServiceDescriptor serviceDescriptor) {
      * Construct new ServerServiceDefinition.
      */
     public ServerServiceDefinition build() {
+      Map<String, ServerMethodDefinition<?, ?>> tmpMethods =
+          new HashMap<String, ServerMethodDefinition<?, ?>>(methods);
+      for (MethodDescriptor<?, ?> descriptorMethod : serviceDescriptor.getMethods()) {
+        ServerMethodDefinition<?, ?> removed = tmpMethods.remove(
```

Comment:
Do we want to make sure the `MethodDescriptors` are the same (reference equality) or is just the name fine? Checking reference equality means we could relax it in the future but seems like it may be a burden if you are making modifications. But, if you are making modifications and not applying it to both sets you may get into trouble. Or maybe not because you added something server-specific. I am a bit concerned of what may go wrong when consumers of these to assume the MethodDescriptors are identical/equivalent.